runtime-attach/src/main/java/io/opentelemetry/contrib/attach/RuntimeAttach.java

```diff
@@ -7,22 +7,82 @@
 
 import java.io.File;
 import java.lang.management.ManagementFactory;
+import java.util.logging.Logger;
 import net.bytebuddy.agent.ByteBuddyAgent;
 
+/** This class allows you to attach the OTel agent for Java at runtime. */
 public final class RuntimeAttach {
 
+  private static final Logger LOGGER = Logger.getLogger(RuntimeAttach.class.getName());
+  private static final String AGENT_ENABLED_PROPERTY = "otel.javaagent.enabled";
+  private static final String AGENT_ENABLED_ENV_VAR = "OTEL_JAVAAGENT_ENABLED";
+  static final String MAIN_THREAD_CHECK_PROP = "otel.javaagent.runtimeattach.mainthreadcheck";
```

Comment:
I think adding `.testing` somewhere helps to make it clear this property is only for running tests (not for end users), we have some precedent for this in instrumentation repo, and for system property names we typically snake-case them, e.g.

```suggestion
  static final String MAIN_THREAD_CHECK_PROP = "otel.javaagent.testing.runtime-attach.main-thread-check";
```